### PR TITLE
docs: libraries BOM principle

### PR DESCRIPTION
**Pull request by suztomo**
Updating README.md to describe what is part of the Libraries BOM and what is not.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,30 +24,84 @@ To use it in Maven, add the following to your POM:
...
+### Core Google dependency libraries
+
+Core Google dependency libraries are the Google-developed Java libraries used by
+the client libraries.
+This includes low-level libraries such as Guava, Protobuf, gRPC, Google Auth
+Libraries.
+These libraries do not have a specific Google Cloud service
+associated to it.
+
+While the users of Google Cloud do not have to directly interact with the core
```
Comment (meltsufin) [inline README.md:55]: I'm not sure that strictly accurate. They may interact with Guava and Auth classes, for example.

File: README.md [hunk 2]
```diff
@@ -22,30 +24,84 @@ To use it in Maven, add the following to your POM:
...
 
-Dependency Dashboard
-------------
+When you use the Libraries BOM, you don't specify individual library
+versions and your application runs on a consistent set of the dependencies.
+
+## Libraries in Scope
+
+The content of the Libraries BOM consists of 2 categories: Google Cloud Java
+client libraries and core Google dependency libraries.
```
Comment (meltsufin) [inline README.md:33]: I think this doesn't answer the question why some client libraries are not in the BOM, such as some of the ones that are not GA.

File: README.md (hunk 1 above)
Comment (suztomo) [reply README.md]: Updated to clarify the meaning.

File: README.md (hunk 2 above)
Comment (suztomo) [reply README.md:33]: Added a section for preview state libraries.

**Comment by suztomo** [comment]
@meltsufin PTAL.

**Pull request closed without merging**